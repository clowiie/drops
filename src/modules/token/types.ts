import Token from 'common/constants/tokens/bsc'

export interface TokenType {
  address: Token
  name: string
  symbol: string
  logo?: string
}
