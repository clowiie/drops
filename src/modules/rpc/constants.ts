export const BSC_RPC_URL = 'https://bsc-dataseed.binance.org'

export const NOT_FOUND_ADDRESS = '0x0000000000000000000000000000000000000000'
