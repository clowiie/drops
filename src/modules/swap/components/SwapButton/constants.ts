export const DISABLE_BUTTON_CLASS =
  'bg-gray-200 text-white rounded-md p-3 text-lg font-semibold pointer-events-none'

export const ACTIVE_BUTTON_CLASS =
  'bg-gray-700 hover:bg-gray-500 text-white rounded-md p-3 text-lg font-semibold text-center'
