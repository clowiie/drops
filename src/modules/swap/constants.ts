export const SWAP_ROOT_PATH = '/'

export const MAXIMUM_UINT256 =
  '0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff'

export const DEFAULT_GAS_LIMIT = '150000'

export enum InputSwapToken {
  In = 'in',
  Out = 'out',
}
