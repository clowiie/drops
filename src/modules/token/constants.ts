import Token from 'common/constants/tokens/bsc'
import { TokenType } from 'modules/token/types'

import ALPACALogo from 'modules/token/images/alpaca-logo.png'
import BUNNYLogo from 'modules/token/images/bunny-logo.svg'
import BUSDLogo from 'modules/token/images/busd-logo.svg'
import CAKELogo from 'modules/token/images/cake-logo.svg'
import ETHLogo from 'modules/token/images/eth-logo.svg'

const tokenList = {
  ALPACA: {
    address: Token.Alpaca,
    name: 'Alpaca',
    symbol: 'ALPACA',
    logo: ALPACALogo,
  },
  BUNNY: {
    address: Token.Bunny,
    name: 'Pancake BUNNY',
    symbol: 'BUNNY',
    logo: BUNNYLogo,
  },
  BUSD: {
    address: Token.Busd,
    name: 'BUSD Token',
    symbol: 'BUSD',
    logo: BUSDLogo,
  },
  CAKE: {
    address: Token.Cake,
    name: 'PancakeSwap Token',
    symbol: 'CAKE',
    logo: CAKELogo,
  },
  ETH: {
    address: Token.Eth,
    name: 'Ethereum Token',
    symbol: 'ETH',
    logo: ETHLogo,
  },
} as Record<string, TokenType>

export const TokenList = Object.freeze(tokenList)
