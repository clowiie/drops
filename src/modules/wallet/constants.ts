import { Connector } from 'modules/wallet/types'

import MetaMaskLogo from 'modules/wallet/images/meta-mask-logo.svg'
import WalletConnectLogo from 'modules/wallet/images/wallet-connect-logo.svg'
import MetaMaskController from 'modules/wallet/stores/controllers/MetaMask'
import WalletConnectController from 'modules/wallet/stores/controllers/WalletConnect'

export const BRIDGE_URL = 'https://pancakeswap.bridge.walletconnect.org/'
export const POLLING_INTERVAL = 12000

export enum MetaMaskMethod {
  GetPermissons = 'wallet_getPermissions',
  RequestAccounts = 'eth_requestAccounts',
  SwitchChain = 'wallet_switchEthereumChain',
}

export enum MetaMaskRPCApi {
  ChainChanged = 'chainChanged',
  AccountsChanged = 'accountsChanged',
}

export enum ConnectorId {
  MetaMask = 'metaMask',
  WalletConnect = 'walletConnect',
}

export enum ConnectorName {
  MetaMask = 'MetaMask',
  WalletConnect = 'WalletConnect',
}

export const connectorList = [
  {
    name: ConnectorName.MetaMask,
    logo: MetaMaskLogo,
    connectorId: ConnectorId.MetaMask,
  },
  {
    name: ConnectorName.WalletConnect,
    logo: WalletConnectLogo,
    connectorId: ConnectorId.WalletConnect,
  },
] as Connector[]

export const ConnectorList = Object.freeze(connectorList)

const walletMapper = {
  [ConnectorId.MetaMask]: MetaMaskController,
  [ConnectorId.WalletConnect]: WalletConnectController,
}

export const WALLET_MAPPER = Object.freeze(walletMapper)
