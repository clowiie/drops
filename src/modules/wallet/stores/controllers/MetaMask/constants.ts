export enum MetaMaskMethod {
    GetPermissons = 'wallet_getPermissions',
    RequestAccounts = 'eth_requestAccounts',
    SwitchChain = 'wallet_switchEthereumChain',
  }

export enum MetaMaskRPCApi {
    ChainChanged = 'chainChanged',
    AccountsChanged = 'accountsChanged',
  }
