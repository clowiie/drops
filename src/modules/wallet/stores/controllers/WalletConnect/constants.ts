export enum WalletConnectorRPCApi {
  ChainChanged = 'chainChanged',
  AccountsChanged = 'accountsChanged',
  Disconnect = 'disconnect',
}
