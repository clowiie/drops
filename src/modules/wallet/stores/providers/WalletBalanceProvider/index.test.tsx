import { render } from '@testing-library/react'

import { TokenList } from 'modules/token/constants'

import * as WalletBalanceProviderType from '.'

describe('WalletBalanceProvider', () => {
  const useGetWalletBalanceSpy = jest.fn()
  jest.doMock(
    'modules/wallet/hooks/useGetWalletBalance',
    () => useGetWalletBalanceSpy,
  )

  const Component = () => <h1>Component</h1>

  const { default: WalletBalanceProvider } =
    require('.') as typeof WalletBalanceProviderType

  beforeEach(() => {
    useGetWalletBalanceSpy.mockReturnValue(TokenList)
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  describe('snapshots', () => {
    it('should match snapshots', () => {
      const { container, getByText } = render(
        <WalletBalanceProvider>
          <Component />
        </WalletBalanceProvider>,
      )

      expect(container.firstChild).toMatchSnapshot()
      expect(getByText('Component')).toBeDefined()
    })
  })
})
