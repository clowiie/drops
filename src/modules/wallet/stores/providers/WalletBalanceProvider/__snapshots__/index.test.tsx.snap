// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletBalanceProvider snapshots should match snapshots 1`] = `
<h1>
  Component
</h1>
`;
